### DISCUSSION TITLE
Performance Benchmarking

### DISCUSSION THREAD

**Discussion by jgollenz** [nvim-orgmode/orgmode#288] [Q&A] [answered]
How would one go about to measure the time each plugin takes when issuing e.g. a vim movement? I get the feeling that, since the recent addition of some treesitter features, this plugin has become noticeably slower for me. What I experience is really janky movement when I spam `j` to move down a ~2.5k lines file with every fold closed. When the folds are open, everything is smooth. So is the general movement with `minimal_init.lua`. When I incrementally throw out my other plugins, it gets better, but it's hard to tell which plugins are the culprits and why. 

What I'm looking for is something like [vim-startuptime](https://github.com/dstein64/vim-startuptime), but more fine-grained. Any ideas?

edit: 

Ok, I tried with https://github.com/nvim-orgmode/orgmode/commit/ba98745a1413df7677b0096866a17ae8fccd08e1, which is before the treesitter features and it moves smooth. There has definitely been some performance penalty somewhere. I'll make an issue

**Comment by kristijanhusak** [comment]
Can you try on this commit https://github.com/nvim-orgmode/orgmode/commit/541f09768591a853309662b015d08e482afe3d72 ? This is before I introduced tree-sitter for folding, maybe that's where the issue happens.

**Reply by jgollenz** [reply]
@kristijanhusak I tried 541f097, but there is no change. It got worse with 8a61c91, this much I can tell. I'm stepping backwards through the commits and will report

**Reply by jgollenz** [reply]
Ok, looks like it completely goes away for me on 88dac80

**Reply by jgollenz** [reply]
The next commit was about extmark. I do use a lot of links and have `conceallevel = 2`

**Reply by kristijanhusak** [reply]
Ok this is probably due to new markup highlighting.

**Comment by kristijanhusak** [comment, marked as answer]
You can do some performance benchmarking like this:

```
:profile start profile.log
:profile file *
:profile func *
```

Do the thing that's slow, and then:
```
:profile stop
:qa!
```

Then open up `profile.log` file that was created
